devfile/devworkspace-operator
# Add support for overriding any pod spec field in DevWorkspace deployment

### What does this PR do?
Add support for specifying Pod spec overrides in DevWorkspaces. This PR is open as a draft for review/discussion on approach. 

The first commit in this PR updates the devfile/api dependency to point at my fork, to pull in changes from PR https://github.com/devfile/api/pull/872. Before merging this PR, the devfile/api PR should be merged and the devfile/api dependency should be updated normally.

### What issues does this PR fix or reference?
Related: https://github.com/devfile/api/issues/860
Closes https://github.com/devfile/devworkspace-operator/issues/852

### Is it tested? How?
To test the changes, there's a new sample:
```
oc apply -f samples/pod-spec-overrides.yaml
```
This workspace will likely fail to start as it sets the scheduler to `stork` and the runtimeClassName to `kata`, which are likely not available. To verify things are working as expected, check that fields are overridden in the Deployment for that workspace

### PR Checklist
